File: CodeChef/DECODEIT.py
t = int(input())

while(t != 0):

    n = int(input())
    binary = input()

    i = 0
    while(i < n):
        temp = binary[i:i+4]
        print(str(int(n, 2) + 97))
        i += 4

    t -= 1

File: CodeChef/DECODEIT.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    long long int t, n, i, j;
    cin >> t;

    while (t--)
    {
        cin >> n;
        char bin[n], temp[4];

        for (i = 0; i < n; i++)
            cin >> bin[i];

        for (i = 0; i < n; i += 4)
        {
            for (j = 0; j < 4; j++)
                temp[j] = bin[i + j];

            cout << (char)(stoi(temp, 0, 2) + 97);
        }
        cout << endl;
    }
    return 0;
}